From 3ae446671b33dd44b94b105978794b086163d45e Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Fri, 9 Nov 2018 14:13:31 +0800
Subject: [PATCH 5070/5242] MLK-20252 ARM64: dts: correct imx8mm root memory

commit  a9c2aa010df8cd5b896f7099f236bf18d48db536 from
https://source.codeaurora.org/external/imx/linux-imx.git

Correct the memory for root linux.
Because rpmsg reserved memory, we could not support 2nd Linux now with
the new memory. So remove the support for 2nd linux.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Reviewed-by: Flynn xu <flynn.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8mm-evk-inmate.dts   |  224 --------------------
 .../boot/dts/freescale/fsl-imx8mm-evk-root.dts     |   65 +-----
 2 files changed, 10 insertions(+), 279 deletions(-)
 delete mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-inmate.dts

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-inmate.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-inmate.dts
deleted file mode 100644
index a6eba06..0000000
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-inmate.dts
+++ /dev/null
@@ -1,224 +0,0 @@
-/*
- * Copyright 2018 NXP
- *
- * This program is free software; you can redistribute it and/or
- * modify it under the terms of the GNU General Public License
- * as published by the Free Software Foundation; either version 2
- * of the License, or (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- */
-
-/dts-v1/;
-
-#include "fsl-imx8mm.dtsi"
-
-/ {
-	model = "Freescale i.MX8MM EVK";
-	compatible = "fsl,imx8mm-evk", "fsl,imx8mm";
-	interrupt-parent = <&gic>;
-
-	timer {
-		compatible = "arm,armv8-timer";
-		interrupts = <GIC_PPI 13 (GIC_CPU_MASK_SIMPLE(6) | IRQ_TYPE_LEVEL_LOW)>, /* Physical Secure */
-			     <GIC_PPI 14 (GIC_CPU_MASK_SIMPLE(6) | IRQ_TYPE_LEVEL_LOW)>, /* Physical Non-Secure */
-			     <GIC_PPI 11 (GIC_CPU_MASK_SIMPLE(6) | IRQ_TYPE_LEVEL_LOW)>, /* Virtual */
-			     <GIC_PPI 10 (GIC_CPU_MASK_SIMPLE(6) | IRQ_TYPE_LEVEL_LOW)>; /* Hypervisor */
-		clock-frequency = <8333333>;
-	};
-
-	clocks {
-		clk_dummy: clock@7 {
-			compatible = "fixed-clock";
-			reg = <7>;
-			#clock-cells = <0>;
-			clock-frequency = <0>;
-			clock-output-names = "clk_dummy";
-		};
-
-		/* The clocks are configured by 1st OS */
-		clk_200m: clock@8 {
-			compatible = "fixed-clock";
-			reg = <8>;
-			#clock-cells = <0>;
-			clock-frequency = <200000000>;
-			clock-output-names = "200m";
-		};
-		clk_266m: clock@9 {
-			compatible = "fixed-clock";
-			reg = <9>;
-			#clock-cells = <0>;
-			clock-frequency = <266000000>;
-			clock-output-names = "266m";
-		};
-		clk_80m: clock@10 {
-			compatible = "fixed-clock";
-			reg = <10>;
-			#clock-cells = <0>;
-			clock-frequency = <80000000>;
-			clock-output-names = "80m";
-		};
-	};
-
-	display-subsystem {
-		/delete-property/ compatible;
-	};
-
-	pci@bfb00000 {
-		compatible = "pci-host-ecam-generic";
-		device_type = "pci";
-		bus-range = <0 0>;
-		#address-cells = <3>;
-		#size-cells = <2>;
-		#interrupt-cells = <1>;
-		interrupt-map-mask = <0 0 0 7>;
-		interrupt-map = <0 0 0 1 &gic GIC_SPI 76 IRQ_TYPE_EDGE_RISING>;
-		reg = <0x0 0x7fb00000 0x0 0x100000>;
-		ranges = <0x02000000 0x00 0x10000000 0x0 0x10000000 0x00 0x10000>;
-	};
-};
-
-/delete-node/ &{/memory@40000000};
-/delete-node/ &{/reserved-memory};
-/delete-node/ &{/busfreq};
-/delete-node/ &ddr_pmu0;
-
-&hsio_pd {
-	status = "disabled";
-};
-
-&pcie0_pd {
-	status = "disabled";
-};
-
-&usb_otg1_pd {
-	status = "disabled";
-};
-
-&usb_otg2_pd {
-	status = "disabled";
-};
-
-&gpumix_pd {
-	status = "disabled";
-};
-
-&vpumix_pd {
-	status = "disabled";
-};
-
-&vpu_g1_pd {
-	status = "disabled";
-};
-
-&vpu_g2_pd {
-	status = "disabled";
-};
-
-&vpu_h1_pd {
-	status = "disabled";
-};
-
-&dispmix_pd {
-	status = "disabled";
-};
-
-&mipi_pd {
-	status = "disabled";
-};
-
-&gpio1 {
-	status = "disabled";
-};
-&gpio2 {
-	status = "disabled";
-};
-&gpio3 {
-	status = "disabled";
-};
-&gpio4 {
-	status = "disabled";
-};
-&gpio5 {
-	status = "disabled";
-};
-
-/delete-node/ &tmu;
-/delete-node/ &{/thermal-zones};
-/delete-node/ &iomuxc;
-
-&gpr {
-	/delete-property/ compatible;
-};
-
-/delete-node/ &anatop;
-/delete-node/ &snvs;
-
-&clk {
-	/delete-property/ compatible;
-};
-
-&src {
-	/delete-property/ compatible;
-};
-
-/delete-node/ &system_counter;
-
-/delete-node/ &imx_rpmsg;
-/delete-node/ &ocotp;
-
-&dispmix_gpr {
-	/delete-property/ compatible;
-};
-
-&sdma1 {
-	status = "disabled";
-};
-
-&sdma2 {
-	status = "disabled";
-};
-
-&sdma3 {
-	status = "disabled";
-};
-
-/delete-node/ &{/imx_ion};
-/delete-node/ &pcie0;
-/delete-node/ &crypto;
-/delete-node/ &caam_sm;
-/delete-node/ &caam_snvs;
-/delete-node/ &irq_sec_vio;
-
-/delete-node/ &{/cpus/cpu@0};
-/delete-node/ &{/cpus/cpu@1};
-/delete-node/ &{/pmu};
-
-&gic {
-	reg = <0x0 0x38800000 0 0x10000>, /* GIC Dist */
-	      <0x0 0x38880000 0 0xC0000>; /* GICR (RD_base + SGI_base) */
-};
-
-&uart4 {
-	clocks = <&osc_24m>,
-		<&osc_24m>;
-	clock-names = "ipg", "per";
-	/delete-property/ dmas;
-	/delete-property/ dmas-names;
-	status = "okay";
-};
-
-&usdhc3 {
-	clocks = <&clk_dummy>,
-		<&clk_266m>,
-		<&clk_200m>;
-	/delete-property/assigned-clocks;
-	/delete-property/assigned-clock-rates;
-	clock-names = "ipg", "ahb", "per";
-	bus-width = <8>;
-	non-removable;
-	status = "okay";
-};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-root.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-root.dts
index ff5d607..1e0aaed 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-root.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-root.dts
@@ -26,33 +26,6 @@
 	/*arm,psci-suspend-param = <0x0>;*/
 };
 
-&clk {
-	init-on-array = <IMX8MM_CLK_AHB_CG IMX8MM_CLK_DRAM_CORE
-			 IMX8MM_CLK_NOC_CG IMX8MM_CLK_NOC_APB_CG
-			 IMX8MM_CLK_USB_BUS_CG
-			 IMX8MM_CLK_MAIN_AXI_CG IMX8MM_CLK_AUDIO_AHB_CG
-			 IMX8MM_CLK_DRAM_APB_DIV IMX8MM_CLK_A53_DIV
-			 IMX8MM_ARM_PLL_OUT IMX8MM_CLK_DISP_AXI_CG
-			 IMX8MM_CLK_DISP_APB_CG
-			 IMX8MM_CLK_NAND_USDHC_BUS_CG>;
-};
-
-&iomuxc {
-	imx8mq-evk {
-		/*
-		 * Used for the 2nd Linux.
-		 * TODO: M4 may use these pins.
-		 */
-		pinctrl_uart4: uart4grp {
-			fsl,pins = <
-				MX8MM_IOMUXC_UART4_RXD_UART4_DCE_RX	0x140
-				MX8MM_IOMUXC_UART4_TXD_UART4_DCE_TX	0x140
-			>;
-		};
-
-	};
-};
-
 &{/busfreq} {
 	/* Disable busfreq, to avoid 1st Linux busfreq crash other inmates */
 	status = "disabled";
@@ -64,46 +37,28 @@
 		reg = <0 0xbfc00000 0x0 0x400000>;
 	};
 
-	inmate_reserved: inmate@0x80000000 {
+	inmate_reserved: inmate@0xbbc00000 {
 		no-map;
-		reg = <0 0x80000000 0x0 0x3fc00000>;
+		reg = <0 0xbbc00000 0x0 0x4000000>;
 	};
 
-	loader_reserved: loader@0x7ff00000 {
+	ivshmem_reserved: ivshmem@0xbbb00000 {
 		no-map;
-		reg = <0 0x7ff00000 0x0 0x00100000>;
+		reg = <0 0xbbb00000 0x0 0x00100000>;
 	};
 
-	ivshmem_reserved: ivshmem@0x7fe00000 {
+	ivshmem2_reserved: ivshmem2@0xbba00000 {
 		no-map;
-		reg = <0 0x7fe00000 0x0 0x00100000>;
+		reg = <0 0xbba00000 0x0 0x00100000>;
 	};
 
-	ivshmem2_reserved: ivshmem2@0x7fd00000 {
+	pci_reserved: pci@0xbb800000 {
 		no-map;
-		reg = <0 0x7fd00000 0x0 0x00100000>;
+		reg = <0 0xbb800000 0x0 0x00200000>;
 	};
 
-	pci_reserved: pci@0x7fb00000 {
+	loader_reserved: loader@0xbb700000 {
 		no-map;
-		reg = <0 0x7fb00000 0x0 0x00200000>;
+		reg = <0 0xbb700000 0x0 0x00100000>;
 	};
 };
-
-&uart2 {
-	/* uart2 is used by the 2nd OS, so configure pin and clk */
-	pinctrl-0 = <&pinctrl_uart2>, <&pinctrl_uart4>;
-	assigned-clocks = <&clk IMX8MM_CLK_UART4_SRC>;
-	assigned-clock-parents = <&clk IMX8MM_CLK_24M>;
-};
-
-&usdhc3 {
-	status = "disabled";
-};
-
-&usdhc2 {
-	/* sdhc3 is used by 2nd linux, configure the pin */
-	pinctrl-0 = <&pinctrl_usdhc3>, <&pinctrl_usdhc2>, <&pinctrl_usdhc2_gpio>;
-	pinctrl-1 = <&pinctrl_usdhc3>, <&pinctrl_usdhc2_100mhz>, <&pinctrl_usdhc2_gpio>;
-	pinctrl-2 = <&pinctrl_usdhc3>, <&pinctrl_usdhc2_200mhz>, <&pinctrl_usdhc2_gpio>;
-};
-- 
1.7.9.5

